### PR TITLE
[AKS] `az aks create/update`: Add new parameter `--enable-vpa` to support enabling vertical pod autoscaler for cluster

### PR DESCRIPTION
**Pull request by huizhifan**
**Related command**

`az aks create --enable vpa`: create cluster with vpa enabled
`az aks update --enable/disable vpa`: update cluster to have/remove vpa
**Description**

[AKS] `az aks create/update`: Add support for VPA(vertical pod autoscaler)
**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py [hunk 1]
```diff
@@ -8695,6 +8695,71 @@ def test_aks_update_with_keda(self, resource_group, resource_group_location):
             self.is_empty(),
         ])
 
+    @AllowLargeResponse()
+    @AKSCustomResourceGroupPreparer(random_name_length=17, name_prefix='clitest', location='westus2')
+    def test_aks_create_with_vpa(self, resource_group, resource_group_location):
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py:8700]: Queued [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=77148984&view=results) to validate the change.

File: src/azure-cli/azure/cli/command_modules/acs/managed_cluster_decorator.py [hunk 2]
```diff
@@ -7093,7 +7100,76 @@ def update_workload_auto_scaler_profile(self, mc: ManagedCluster) -> ManagedClus
...
+            if mc.workload_auto_scaler_profile is None:
+                mc.workload_auto_scaler_profile = self.models.ManagedClusterWorkloadAutoScalerProfile()
+            if mc.workload_auto_scaler_profile.vertical_pod_autoscaler is None:
+                mc.workload_auto_scaler_profile.vertical_pod_autoscaler = self.models.ManagedClusterWorkloadAutoScalerProfileVerticalPodAutoscaler()
+
+            # set disabled
+            mc.workload_auto_scaler_profile.vertical_pod_autoscaler.enabled = False
+
         return mc
+    def _get_enable_vpa(self, enable_validation: bool = False) -> bool:
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/managed_cluster_decorator.py:7122]: The getters should be placed under the Context Class.

Comment (huizhifan) [reply src/azure-cli/azure/cli/command_modules/acs/managed_cluster_decorator.py]: done

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py (hunk 1 above)
Comment (FumingZhang) [reply src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py:8700]: Queued [new runs](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=77465557&view=results), please commit the recordings files (find from artifact) if test passed.

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py [hunk 3]
```diff
@@ -8695,6 +8695,75 @@ def test_aks_update_with_keda(self, resource_group, resource_group_location):
             self.is_empty(),
         ])
 
+    @live_only()
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py:8698]: Please remove this and upload recording file

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py [hunk 4]
```diff
@@ -8695,6 +8695,75 @@ def test_aks_update_with_keda(self, resource_group, resource_group_location):
...
+            self.check('workloadAutoScalerProfile.verticalPodAutoscaler.enabled', True),
+        ])
+
+        # delete
+        delete_cmd = 'aks delete --resource-group={resource_group} --name={name} --yes --no-wait'
+        self.cmd(delete_cmd, checks=[
+            self.is_empty(),
+        ])
+
+    @live_only()
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py:8725]: Please remove this and upload recording file

**Review by FumingZhang — APPROVED** [review]
LGTM

**Comment by FumingZhang** [comment]
The failure of the CI is caused by some other change, maybe this one #27036?

**Comment by evelyn-ys** [comment]
> The failure of the CI is caused by some other change, maybe this one #27036?

Yes, RDBMS contributor is fixing this test

**Review by FumingZhang — APPROVED** [review]
LGTM

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**